### PR TITLE
Add tests for Story_Post_Type class

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add some more tests to stories class. 

## Testing Instructions

---

See #328.

### CODE REVIEW

File: tests/phpunit/tests/Story_Post_Type.php [hunk 1]
```diff
@@ -109,4 +126,68 @@ public function test_get_post_type_icon() {
...
+		$this->assertTrue( wp_style_is( \Google\Web_Stories\Story_Post_Type::WEB_STORIES_SCRIPT_HANDLE, 'registered' ) );
+	}
+
+	/**
+	 * @covers \Google\Web_Stories\Story_Post_Type::filter_site_kit_gtag_opt
+	 */
+	public function test_filter_site_kit_gtag_opt() {
+		global $wp_query;
+		$wp_query->is_singular    = true;
+		$wp_query->queried_object = get_post( self::$story_id );
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php]: I think you can just do `$this->go_to( get_permalink( self::$story_id ) )` to set the current query correctly. That's what core does.

File: includes/Story_Post_Type.php [hunk 2]
```diff
@@ -501,7 +501,7 @@ public function filter_site_kit_gtag_opt( $gtag_opt ) {
 			return $gtag_opt;
 		}
 
-		$post = get_post();
+		$post = get_queried_object();
```
Comment (swissspidy) [inline includes/Story_Post_Type.php]: Can you elaborate on this change?

File: tests/phpunit/tests/Story_Post_Type.php (hunk 1 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/Story_Post_Type.php]: Fixed

File: includes/Story_Post_Type.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php]: I changed it, as seemed cleaner. But I will change it back.

File: tests/phpunit/tests/Story_Post_Type.php [hunk 3]
```diff
@@ -109,4 +126,85 @@ public function test_get_post_type_icon() {
...
 	}
+
+	/**
+	 * @covers \Google\Web_Stories\Story_Post_Type::admin_enqueue_scripts
+	 */
+	public function test_admin_enqueue_scripts() {
+		$post_type_object = new \Google\Web_Stories\Story_Post_Type();
+		set_current_screen( 'post.php' );
+		get_current_screen()->post_type = \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG;
+		get_current_screen()->base      = 'post';
```
Comment (swissspidy) [inline tests/phpunit/tests/Story_Post_Type.php]: Same optimization as in #2849:

```suggestion
		$GLOBALS['current_screen'] = convert_to_screen( \Google\Web_Stories\Story_Post_Type::POST_TYPE_SLUG );
```

**Review by swissspidy — APPROVED** [review]
Approved with 1 nit

**Pull request merged**